x-pack/legacy/plugins/uptime/server/rest_api/monitors/monitors_details.ts

```diff
@@ -3,23 +3,27 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import Joi from 'joi';
+
+import { schema } from '@kbn/config-schema';
 import { UMServerLibs } from '../../lib/lib';
-import { MonitorDetails } from '../../../common/runtime_types/monitor/monitor_details';
+import { UMRestApiRouteCreator } from '../types';
 
-export const createGetMonitorDetailsRoute = (libs: UMServerLibs) => ({
+export const createGetMonitorDetailsRoute: UMRestApiRouteCreator = (libs: UMServerLibs) => ({
   method: 'GET',
   path: '/api/uptime/monitor/details',
+  validate: {
+    query: schema.object({
+      monitorId: schema.maybe(schema.string()),
+    }),
+  },
   options: {
-    validate: {
-      query: Joi.object({
-        monitorId: Joi.string(),
-      }),
-    },
     tags: ['access:uptime'],
   },
-  handler: async (request: any): Promise<MonitorDetails> => {
+  handler: async (_context, request, response): Promise<any> => {
     const { monitorId } = request.query;
-    return await libs.monitors.getMonitorDetails(request, monitorId);
+
+    return response.ok({
+      body: { ...(await libs.monitors.getMonitorDetails(request, monitorId)) },
+    });
   },
```

Comment:
should we be doing try/catch for all routes? just like it's done in get_index_pattern?

Reply:
Probably. I had included a try...catch with `get_index_pattern` primarily because it was touching saved objects, which I didn't have much experience with. Now that Kibana has such an opinionated way to respond to requests it probably wouldn't hurt to define explicit response codes for various errors users could encounter. I'd add that as a follow-up issue though as I feel it's out of scope for what we're doing here.